src/main/java/org/threadly/litesockets/Client.java

```diff
@@ -3,557 +3,358 @@
-   * 
-   * @return the ThreadExecuter for this client.  
-   */
-  protected SubmitterExecutorInterface getThreadExecuter() {
-    return sei;
-  }
-
-  /**
-   * Sets this client to the specified SocketExecuter.  This happens when the client is added to the SocketExecuter.
-   * This method should not be overridden. 
+   * <p> This returns this clients {@link Executor}.  The client must have been added to the {@link SocketExecuterInterface} or 
+   * this will return null.</p>
    * 
+   * <p> Its worth noting that operations done on this {@link Executor} can/will block Read callbacks on the 
+   * client, but it does provide you the ability to execute things on the clients read thread</p>
    * 
-   * @param ce the SocketExecuter to set for this client.
+   * @return The {@link Executor} for the client.
    */
-  protected void setSocketExecuter(SocketExecuterBase ce) {
-    this.ce = ce;
-  }
+  public Executor getClientsThreadExecutor();
   
   /**
-   * This is used to get the clients currently assigned SocketExecuter.
+   * <p>This is set when the client is added to a {@link SocketExecuterInterface}.  Care should be given if you manually set this as 
+   * it will greatly impact the behavior of the client {@link Reader} callback.<p>
    * 
-   * @return the SocketExecuter set for this client. if none, null is returned.
+   * @param cte the {@link Executor} to used for this clients callbacks.
    */
-  protected SocketExecuterBase getSocketExecuter() {
-    return ce;
-  }
+  public void setClientsThreadExecutor(Executor cte);
   
   /**
-   * Used by extending classes to notify the SocketExecuter when this client can Read again.
+   * <p>This is used to get the clients currently assigned {@link SocketExecuterInterface}.</p>
+   * 
+   * @return the {@link SocketExecuterInterface} set for this client. if none, null is returned.
    */
-  protected void flagReadable() {
-    if(ce != null) {
-      ce.flagNewRead(this);
-    }
-  }
+  public SocketExecuterInterface getClientsSocketExecuter();
   
   /**
-   * Used by extending classes to notify the SocketExecuter when this client can write again.
-   */
-  protected void flagWriteable() {
-    if(ce != null) {
-      ce.flagNewWrite(this);
-    }
-  }
-  
-  /**
-   * The SocketExecuter calls this method when it detects a socket close event. 
-   * This notifies the clients Closer if set.
+   * <p>This is automatically done when called the client is added with {@link SocketExecuterInterface#addClient(Client)}.  
+   * In general there is no need to manually set this, and care should be given if done.</p>
+   * 
+   * @param cse the {@link SocketExecuterInterface} for this client.
    */
-  protected void callCloser() {
-    if(sei != null && closer != null) {
-      sei.execute(new Runnable() {
-        @Override
-        public void run() {
-          getCloser().onClose(Client.this);
-        }});
-    }
-  }
+  public void setClientsSocketExecuter(SocketExecuterInterface cse);
```

Comment:
Thats what I was doing before when the Client was an abstract class.  I would love to find a way to either hide or not need these setters on the client and server since its really up to the implementation if they even need them.